File: samples/vulnerable-sample.js
// Sample vulnerable JavaScript code for testing Scanax
// This file contains intentional security vulnerabilities for demonstration

const express = require('express');
const mysql = require('mysql');
const app = express();

// VULNERABILITY 1: Hardcoded Database Credentials
const db = mysql.createConnection({
    host: 'localhost',
    user: 'root',
    password: 'SuperSecret123!',  // Hardcoded password
    database: 'production_db'
});

// VULNERABILITY 2: Hardcoded API Key
const API_KEY = "sk-1234567890abcdefghijklmnop";
const AWS_SECRET = "aws_secret_access_key_AKIAIOSFODNN7EXAMPLE";

// VULNERABILITY 3: SQL Injection
app.get('/user/:id', (req, res) => {
    const userId = req.params.id;
    const sql = 'SELECT * FROM users WHERE id = ' + userId;  // String concatenation
    db.query(sql, (err, result) => {
        if (err) return res.status(500).send(err);
        res.send(result);
    });
});

// VULNERABILITY 4: SQL Injection with user input
app.get('/search', (req, res) => {
    const username = req.query.name;
    const sql = "SELECT * FROM users WHERE username = '" + username + "'";
    db.query(sql, (err, result) => {
        if (err) return res.status(500).send(err);
        res.send(result);
    });
});

// VULNERABILITY 5: Cross-Site Scripting (XSS)
app.get('/greet', (req, res) => {
    const name = req.query.name;
    res.send('<h1>Hello ' + name + '!</h1>');  // Unescaped user input
});

// VULNERABILITY 6: Command Injection
const { exec } = require('child_process');
app.get('/ping', (req, res) => {
    const host = req.query.host;
    exec('ping ' + host, (error, stdout) => {  // Unsafe command execution
        res.send(stdout);
    });
});

// VULNERABILITY 7: Path Traversal
const fs = require('fs');
app.get('/file', (req, res) => {
    const filename = req.query.name;
    fs.readFile('/uploads/' + filename, (err, data) => {  // No path validation
        if (err) return res.status(404).send('Not found');
        res.send(data);
    });
});

// VULNERABILITY 8: Weak Cryptography
const crypto = require('crypto');
function hashPassword(password) {
    return crypto.createHash('md5').update(password).digest('hex');  // MD5 is weak
}

// VULNERABILITY 9: eval() usage
app.post('/calculate', (req, res) => {
    const expression = req.body.expr;
    const result = eval(expression);  // Dangerous eval()
    res.json({ result });
});

// VULNERABILITY 10: Missing Authentication
app.get('/admin/users', (req, res) => {
    // No authentication check!
    db.query('SELECT * FROM users', (err, users) => {
        res.json(users);
    });
});

app.listen(3000, () => {
    console.log('Server running on http://localhost:3000');
});

/* 
 * Try scanning this file with Scanax!
 * Press Ctrl+Shift+S to scan
 * Hover over red squiggles to see vulnerability details
 * Click "Get Fix Suggestion" to see secure code alternatives
 */

File: samples/vulnerable-sample.py
# Sample vulnerable Python code for testing Scanax

import sqlite3
import os
import pickle
from flask import Flask, request

app = Flask(__name__)

# VULNERABILITY 1: Hardcoded Secret Key
app.secret_key = 'hard-coded-secret-key-12345'

# VULNERABILITY 2: Debug Mode Enabled
app.config['DEBUG'] = True

# VULNERABILITY 3: Hardcoded Database Credentials
DB_USER = 'admin'
DB_PASSWORD = 'SuperSecret123!'
DATABASE_URL = f'postgresql://{DB_USER}:{DB_PASSWORD}@localhost/prod_db'

# VULNERABILITY 4: SQL Injection
@app.route('/user/<user_id>')
def get_user(user_id):
    conn = sqlite3.connect('database.db')
    cursor = conn.cursor()
    query = f"SELECT * FROM users WHERE id = {user_id}"  # String formatting
    cursor.execute(query)
    return str(cursor.fetchall())

# VULNERABILITY 5: SQL Injection with user input
@app.route('/search')
def search():
    name = request.args.get('name')
    conn = sqlite3.connect('database.db')
    cursor = conn.cursor()
    query = f"SELECT * FROM users WHERE name = '{name}'"
    cursor.execute(query)
    return str(cursor.fetchall())

# VULNERABILITY 6: Command Injection
@app.route('/ping')
def ping():
    host = request.args.get('host')
    result = os.system(f'ping {host}')  # Unsafe command execution
    return f'Result: {result}'

# VULNERABILITY 7: Path Traversal
@app.route('/read')
def read_file():
    filename = request.args.get('file')
    with open(f'/uploads/{filename}', 'r') as f:  # No validation
        return f.read()

# VULNERABILITY 8: Insecure Deserialization
@app.route('/load')
def load_data():
    data = request.args.get('data')
    obj = pickle.loads(data.encode())  # Unsafe pickle.loads()
    return str(obj)

# VULNERABILITY 9: eval() usage
@app.route('/calc')
def calculate():
    expr = request.args.get('expr')
    result = eval(expr)  # Dangerous eval()
    return str(result)

# VULNERABILITY 10: Weak Cryptography
import hashlib
def hash_password(password):
    return hashlib.md5(password.encode()).hexdigest()  # MD5 is weak

# VULNERABILITY 11: Hardcoded API Keys
AWS_ACCESS_KEY = 'AKIAIOSFODNN7EXAMPLE'
AWS_SECRET_KEY = 'wJalrXUtnFEMI/K7MDENG/bPxRfiCYEXAMPLEKEY'
STRIPE_KEY = 'sk_live_51234567890abcdefghijk'

# VULNERABILITY 12: Missing Authentication
@app.route('/admin/delete/<user_id>')
def delete_user(user_id):
    # No authentication check!
    conn = sqlite3.connect('database.db')
    cursor = conn.cursor()
    cursor.execute(f"DELETE FROM users WHERE id = {user_id}")
    conn.commit()
    return 'User deleted'

if __name__ == '__main__':
    app.run(host='0.0.0.0', port=5000)

"""
Try scanning this file with Scanax!
Press Ctrl+Shift+S to scan
Hover over red squiggles to see vulnerability details
Click "Get Fix Suggestion" to see secure code alternatives
"""
